pdf/model/writer.go

```diff
@@ -112,9 +185,29 @@ func NewPdfWriter() PdfWriter {
 
 	// Creation info.
 	infoDict := core.MakeDict()
-	infoDict.Set("Producer", core.MakeString(getPdfProducer()))
-	infoDict.Set("Creator", core.MakeString(getPdfCreator()))
-	infoObj := core.PdfIndirectObject{}
+	metadata := map[core.PdfObjectName]string{
+		"Producer": getPdfProducer(),
+		"Creator":  getPdfCreator(),
+		"Author":   getPdfAuthor(),
+		"Subject":  getPdfSubject(),
+		"Title":    getPdfTitle(),
+		"Keywords": getPdfKeywords(),
+	}
+	for key, value := range metadata {
+		if value != "" {
+			infoDict.Set(key, core.MakeString(value))
+		}
+	}
+
+	// Set creation and modified dates.
+	if cd, err := NewPdfDateFromTime(getPdfCreationDate()); err == nil {
+		infoDict.Set("CreationDate", cd.ToPdfObject())
+	}
+	if md, err := NewPdfDateFromTime(getPdfModifiedDate()); err == nil {
+		infoDict.Set("ModDate", md.ToPdfObject())
+	}
+
+  infoObj := core.PdfIndirectObject{}
```

Comment:
Wrong indent. Maybe run `gofmt` on the file.

Reply:
Done